tests/test-desktop-files.lua

```diff
@@ -0,0 +1,17 @@
+local runner = require("_runner")
+local menubar = require("menubar")
+local gears_debug = require("gears.debug")
+
+local assets_dir = os.getenv("TEST_ASSETS") .. "/"
+menubar.menu_gen.all_menu_dirs = { assets_dir }
+
+runner.run_steps({
+    function()
+        print("Running desktop file test!")
+        menubar.menu_gen.generate(function(menu_entries)
+            print("Menu list generated!")
+            gears_debug.dump(menu_entries, "menu_entries", 4)
+        end)
```

Comment:
This bit's async - meaning the code here isn't getting run by the time the test finishes. Is there some way around this?